### PR TITLE
Simplify NO_* options

### PR DESCRIPTION
**Pull request by kripken**
We have a few options starting with `NO_`, such as `NO_EXIT_RUNTIME`. The double negative when writing something like `NO_EXIT_RUNTIME=0` can be confusing. This PR simplifies things by making `NO_` an optional prefix that flips a boolean option, that is,

`-s EXIT_RUNTIME=1`

is the same as

`-s NO_EXIT_RUNTIME=0`.

As this is optional, no breakage should occur for existing users.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -789,19 +808,12 @@ def detect_fixed_language_mode(args):
...
-          'BINARYEN_MEM_MAX': 'WASM_MEM_MAX',
-          # TODO: change most (all?) other BINARYEN* names to WASM*
-      }
       settings_key_changes = set()
 
       def setting_sub(s):
         key, value = s.split('=', 1)
         settings_key_changes.add(key)
-        return '='.join([settings_aliases.get(key, key), value])
+        return '='.join([key, value])
```
Comment (sbc100) [inline emcc.py]: This function just returns its first argument now.  Might be worth removing it and replacing the list comprehension below with a loop?

File: site/build/text/docs/tools_reference/emcc.txt [hunk 2]
```diff
@@ -96,6 +96,10 @@ Options that are modified or new in *emcc* are listed below:
    JavaScript code generation option passed into the Emscripten
    compiler. For the available options, see src/settings.js.
 
+   Note: You can prefix boolean options with "NO_" to reverse them.
+     For example, "-s EXIT_RUNTIME=1" is the same as "--s
+     NO_EXIT_RUNTIME=0".
```
Comment (sbc100) [inline site/build/text/docs/tools_reference/emcc.txt]: Only one `-`.

File: site/source/docs/tools_reference/emcc.rst [hunk 3]
```diff
@@ -87,6 +87,8 @@ Options that are modified or new in *emcc* are listed below:
 ``-s OPTION=VALUE``
 	JavaScript code generation option passed into the Emscripten compiler. For the available options, see `src/settings.js <https://github.com/kripken/emscripten/blob/master/src/settings.js>`_. 
 	
+	.. note:: You can prefix boolean options with ``NO_`` to reverse them. For example, ``-s EXIT_RUNTIME=1`` is the same as ``--s NO_EXIT_RUNTIME=0``.
+
```
Comment (sbc100) [inline site/source/docs/tools_reference/emcc.rst]: Double `-`

**Comment by kripken** [comment]
Good idea about dropping `=1`, so that `-s ASSERTIONS` and `-s NO_ASSERTIONS` would work. I think that should be easy to add, only issue I can think of is that `-s` may be ambiguous and we currently check if the item after it has a `=` iirc. But maybe we can simplify that or work around it.

**Pull request merged**